src/background_music/CircularBuffer.h

```diff
@@ -0,0 +1,119 @@
+#ifndef AUDACIOUS_PLUGINS_CIRCULARBUFFER_H
+#define AUDACIOUS_PLUGINS_CIRCULARBUFFER_H
+/*
+ * audacious-plugins/CircularBuffer.h
+ *
+ * Added by michel on 2023-09-24
+ * Copyright (C) 2015-2023 Michel Fleur.
+ * Source https://github.com/emmef/audacious-plugins
+ * Email audacious-plugins@emmef.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils.h"
+
+template<typename T>
+class CircularBuffer
```

Comment:
We already have a ring buffer container in <libaudcore/ringbuf.h>. Let's not add another.